### DISCUSSION TITLE
Undefined variable $component (View: /var/www/media/resources/views/livewire/pages/landing.blade.php)

### DISCUSSION THREAD

**Discussion by axeljerabek** [LycheeOrg/Lychee#2390] [Bugs] [answered]
### Lychee version

v5.2.1

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

[ErrorException](https://media.jerabek.fi/landing#trace-box-2)  [ViewException](https://media.jerabek.fi/landing#trace-box-1)
HTTP 500 Internal Server Error
Undefined variable $component (View: /var/www/media/resources/views/livewire/pages/landing.blade.php)
Exceptions 2
Illuminate\View\
ViewException
Show exception properties
in [/var/www/media/storage/framework/views/14c82ce716a562200459847ac87fcee6.php ](file:///var/www/media/storage/framework/views/14c82ce716a562200459847ac87fcee6.php#L54)(line 54)
    </div>    @component('App\View\Components\Footer', 'footer', ['layout' => 'footer-landing'])<?php if (isset($attributes) && $attributes instanceof Illuminate\View\ComponentAttributeBag && $constructor = (new ReflectionClass(App\View\Components\Footer::class))->getConstructor()): ?><?php $attributes = $attributes->except(collect($constructor->getParameters())->map->getName()->all()); ?><?php endif; ?><?php $component->withAttributes([]); ?>@endComponentClass</div> ?><?php /**PATH /var/www/media/resources/views/livewire/pages/landing.blade.php ENDPATH**/ ?>
in [/var/www/media/vendor/livewire/livewire/src/Mechanisms/ExtendBlade/ExtendedCompilerEngine.php ](file:///var/www/media/vendor/livewire/livewire/src/Mechanisms/ExtendBlade/ExtendedCompilerEngine.php#L58)-> handleViewException (line 58)
            \Illuminate\View\Engines\PhpEngine::handleViewException($e, $obLevel);            return;        }        parent::handleViewException($e, $obLevel);    }    public function shouldBypassExceptionForLivewire(\Throwable $e, $obLevel)    {        $uses = array_flip(class_uses_recursive($e));
in [/var/www/media/vendor/livewire/livewire/src/Mechanisms/ExtendBlade/ExtendedCompilerEngine.php ](file:///var/www/media/vendor/livewire/livewire/src/Mechanisms/ExtendBlade/ExtendedCompilerEngine.php#L40)-> handleViewException (line 40)
            \Closure::bind(function () use ($__path, $__data) {                extract($__data, EXTR_SKIP);                include $__path;            }, $component, $component)();        } catch (\Exception|\Throwable $e) {            $this->handleViewException($e, $obLevel);        }        return ltrim(ob_get_clean());    }
in [/var/www/media/vendor/laravel/framework/src/Illuminate/View/Engines/CompilerEngine.php ](file:///var/www/media/vendor/laravel/framework/src/Illuminate/View/Engines/CompilerEngine.php#L72)-> evaluatePath (line 72)
        // Once we have the path to the compiled file, we will evaluate the paths with        // typical PHP just like any other templates. We also keep a stack of views        // which have been rendered for right exception messages to be generated.        try {            $results = $this->evaluatePath($this->compiler->getCompiledPath($path), $data);        } catch (ViewException $e) {            if (! str($e->getMessage())->contains(['No such file or directory', 'File does not exist at path'])) {                throw $e;            }

### Steps to reproduce the issue

1. Instll or update the current git
2. configure .env
3. load the webpage

### Diagnostics [REQUIRED]

Diagnostics
-----------
Warning: /v*****************ic/uploads/ is owned by group www-data, but should be owned by one out of root
Warning: /v*****************ic/uploads/livewire-tmp is owned by group www-data, but should be owned by one out of root
Warning: /v*****************ic/uploads/small2x is owned by group www-data, but should be owned by one out of root
Warning: /v*****************ic/uploads/small2x/8a is owned by group www-data, but should be owned by one out of root
Warning: /v*****************ic/uploads/small2x/8a/ae is owned by group www-data, but should be owned by one out of root
Warning: 16540 more directories with wrong owner
Warning: Dropbox import not working. dropbox_key is empty.
Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
Warning: zend.assertions is disabled although Lychee is in debug mode. For easier debugging code generation for assertions should be enabled.

System Information
------------------
Lychee Version (git):                    master (74d672a) -- Up to date (2 minutes ago).
DB Version:                              5.2.0

composer install:                        --no-dev
APP_ENV:                                 production
APP_DEBUG:                               true
APP_URL:                                 set
APP_DIR:                                 default
LOG_VIEWER_ENABLED:                      true
LIVEWIRE_ENABLED:                        true

System:                                  Linux
PHP Version:                             8.3.6
PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                Europe/Berlin
Max uploaded file size:                  2M
Max post size:                           8M
Livewire chunk size:                     0.00 B
Max execution time:                      0
MySQL Version:                           10.11.7-MariaDB-2ubuntu2-log

exec() Available:                        yes
Imagick Available:                       1
Imagick Enabled:                         1
Imagick Version:                         1692
GD Version:                              2.3.3
Number of foreign key:                   12 found.

Config Information
------------------
version:                                 050200
check_for_updates:                       1
sorting_photos_col:                      taken_at
sorting_photos_order:                    DESC
sorting_albums_col:                      max_taken_at
sorting_albums_order:                    DESC
imagick:                                 1
skip_duplicates:                         0
small_max_width:                         0
small_max_height:                        360
medium_max_width:                        3840
medium_max_height:                       2160
lang:                                    en
image_overlay_type:                      exif
default_license:                         reserved
compression_quality:                     90
grants_full_photo_access:                1
delete_imported:                         0
mod_frame_enabled:                       1
mod_frame_refresh:                       30
thumb_2x:                                1
small_2x:                                1
medium_2x:                               1
landing_page_enable:                     1
force_migration_in_production:           1
search_public:                           1
hide_version_number:                     0
SL_enable:                               0
SL_for_admin:                            0
recent_age:                              1
grants_download:                         1
photos_wraparound:                       1
map_display:                             1
zip64:                                   1
map_display_public:                      1
map_provider:                            RRZE
force_32bit_ids:                         0
map_include_subalbums:                   1
update_check_every_days:                 3
has_exiftool:                            1
share_button_visible:                    1
import_via_symlink:                      0
has_ffmpeg:                              1
location_decoding:                       1
location_decoding_timeout:               30
location_show:                           1
location_show_public:                    1
rss_enable:                              0
rss_recent_days:                         7
rss_max_items:                           100
prefer_available_xmp_metadata:           1
editor_enabled:                          1
lossless_optimization:                   0
swipe_tolerance_x:                       150
swipe_tolerance_y:                       250
local_takestamp_video_formats:           .avi|.mov
log_max_num_line:                        1000
unlock_password_photos_with_url_param:   0
nsfw_visible:                            1
nsfw_blur:                               1
nsfw_warning:                            1
nsfw_warning_admin:                      1
nsfw_banner_override:
map_display_direction:                   1
album_subtitle_type:                     oldstyle
upload_processing_limit:                 4
new_photos_notification:                 0
legacy_id_redirection:                   1
zip_deflate_level:                       6
SA_enabled:                              1
default_album_protection:                1
allow_username_change:                   1
album_decoration:                        layers
album_decoration_orientation:            row
auto_fix_orientation:                    1
use_job_queues:                          0
random_album_id:                         starred
use_last_modified_date_when_no_exif_date: 0
layout:                                  justified
date_format_photo_thumb:                 M j, Y, g:i:s A e
date_format_photo_overlay:               M j, Y, g:i:s A e
date_format_sidebar_uploaded:            M j, Y, g:i:s A e
date_format_sidebar_taken_at:            M j, Y, g:i:s A e
date_format_hero_min_max:                F Y
date_format_hero_created_at:             M j, Y, g:i:s A T
date_format_album_thumb:                 M Y
upload_chunk_size:                       0
nsfw_banner_blur_backdrop:               0
search_pagination_limit:                 1000
search_minimum_length_required:          4
photo_layout_justified_row_height:       320
photo_layout_masonry_column_width:       300
photo_layout_grid_column_width:          250
photo_layout_square_column_width:        200
photo_layout_gap:                        12
display_thumb_album_overlay:             always
display_thumb_photo_overlay:             hover
use_album_compact_header:                0
default_album_thumb_aspect_ratio:        1/1
login_button_position:                   left
back_button_enabled:                     0
back_button_text:                        Return to Home
current_job_processing_visible:          1

### Browser & System [REQUIRED]

Ubuntu 22.04 LTS and 24.04 Beta
Browser Google Chrome Version 123.0.6312.107

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
WTF.

That is the only thing I can think of in front of that bug.

**Reply by axeljerabek** [reply]
No idea. But I had that bug when I did a "git pull" on the Ubuntu 22.04 variant. So it is not related to the Ubuntu 24.04. Also PHP did not upgrade. Actually nothing else was altered.

**Reply by ildyria** [reply]
after `git pull` did you do `composer install --no-dev`  ?

**Reply by axeljerabek** [reply]
git pull
composer install --no-dev
php artisan migrate
npm install
npm run build

All of these. But also a complete new install does not work and gives the same error. From git or zip, does not matter.

**Reply by ildyria** [reply]
I will investigate that ASAP because this is seriously worrying.

**Reply by axeljerabek** [reply]
Thanks! Looking forward to your findings.

**Reply by ildyria** [reply, marked as answer]
Can you comment line 50 of `/var/www/media/resources/views/livewire/pages/landing.blade.php`  ?

```blade
    <x-footer layout="footer-landing" />
```
to:
```blade
    {{-- <x-footer layout="footer-landing" /> --}}
```

**Reply by axeljerabek** [reply]
Ah, that worked! Thanks a lot! :D
Everything is online again and works like a charm!

**Reply by ildyria** [reply]
it is just a work around.
